### DISCUSSION TITLE
Как вы сделали авторизацию в вк без открытия браузера?

### DISCUSSION THREAD

**Discussion by demidko** [vknet/vk#1251] [Q&A] [answered]
Я пишу на Java/Kotlin, хотелось бы найти аналогичную библиотеку или запилить самому..

**Comment by inyutin-maxim** [comment, marked as answer]
Методом симуляции работы браузера, посредством HTTP запросов

**Comment by HELLRAiSER555** [comment]
> Я пишу на Java/Kotlin, хотелось бы найти аналогичную библиотеку или запилить самому..

Можно использовать элементы управления webBrowser или webView2 со свойством Visible = False, вставлять данные (логин/пароль) в веб форму и нажимать кнопку сабмит, получая таким образом в url token. Можно использовать WebClient или HTTPClient и все те же HTTP запросы. А самый простой и правильный описан тут [Авторизация на базе протокола OAuth](https://dev.vk.com/ru/reference)
